From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Sun, 25 Dec 2022 10:47:29 +0100
Subject: [PATCH] Skip secondary POI sensor if absent

License: LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
Gale - https://galemc.org

This patch is based on the following mixin:
"net/caffeinemc/mods/lithium/mixin/ai/sensor/secondary_poi/SecondaryPoiSensorMixin.java"
By: 2No2Name <2No2Name@web.de>
As part of: Lithium (https://github.com/CaffeineMC/lithium)
Licensed under: LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)

* Lithium description *

villagers that have no secondary POI (farmland) do not search for it

diff --git a/net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java b/net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
index dc4a9ddb8479e9d0c4895b19d7d677cdd8ad3faa..572a37676aebb517220fdbb94d3ca76a203936da 100644
--- a/net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
+++ b/net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
@@ -24,7 +24,10 @@ public class SecondaryPoiSensor extends Sensor<Villager> {
     protected void doTick(ServerLevel level, Villager entity) {
         // Purpur start - Option for Villager Clerics to farm Nether Wart - make sure clerics don't wander to soul sand when the option is off
         Brain<?> brain = entity.getBrain();
-        if (!level.purpurConfig.villagerClericsFarmWarts && entity.getVillagerData().profession().is(net.minecraft.world.entity.npc.VillagerProfession.CLERIC)) {
+        // Gale start - Lithium - skip secondary POI sensor if absent
+        var secondaryPoi = entity.getVillagerData().profession().value().secondaryPoi();
+        if (secondaryPoi.isEmpty() || (!level.purpurConfig.villagerClericsFarmWarts && entity.getVillagerData().profession().is(net.minecraft.world.entity.npc.VillagerProfession.CLERIC))) {
+            // Gale end - Lithium - skip secondary POI sensor if absent
             brain.eraseMemory(MemoryModuleType.SECONDARY_JOB_SITE);
             return;
         }
@@ -38,7 +41,7 @@ public class SecondaryPoiSensor extends Sensor<Villager> {
             for (int i2 = -2; i2 <= 2; i2++) {
                 for (int i3 = -4; i3 <= 4; i3++) {
                     BlockPos blockPos1 = blockPos.offset(i1, i2, i3);
-                    if (entity.getVillagerData().profession().value().secondaryPoi().contains(level.getBlockState(blockPos1).getBlock())) {
+                    if (secondaryPoi.contains(level.getBlockState(blockPos1).getBlock())) { // Gale - Lithium - skip secondary POI sensor if absent
                         list.add(GlobalPos.of(resourceKey, blockPos1));
                     }
                 }
